src/NumericalAlgorithms/Interpolation/MultiLinearSpanInterpolation.hpp

```diff
@@ -0,0 +1,387 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <memory>
+
+#include "DataStructures/Index.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace intrp {
+
+/*!
+ * \brief Performs linear interpolation in arbitrary dimensions.
+ *  The class is non-owning and expects a C-ordered array, (n, x, y, z).
+ *  The variable index, n, varies fastest in memory.
+ *  Note that this class is intentionally non-pupable.
+ *
+ *  \tparam Dimension dimensionality of the table
+ *  \tparam NumberOfVariables number of variables stored in the table
+ *  \tparam UniformSpacing indicated whether the table has uniform _spacing or
+ * not. This is useful for performance reasons in finding the appropriate table
+ * index.
+ */
+
+template <size_t Dimension, size_t NumberOfVariables, bool UniformSpacing>
+class MultiLinearSpanInterpolation {
+
+ public:
+  template <size_t ThisDimension>
+  struct Weight {
+    std::array<double, two_to_the(ThisDimension)> weights;
+    Index<two_to_the(ThisDimension)> index;
+  };
+
+  size_t extents(const size_t which_dimension) const {
+    return _number_of_points[which_dimension];
+  }
+
+  void extrapolateAboveData(size_t which_dimension, const bool value) {
```

Comment:
`const size_t`?

also `snake_case` the function names: `extrapolate_above_data`, `extrapolate_below_data`, `get_weights`, `find_index`, `find_index_general`, `find_index_uniform`